src/tsdb.c

```diff
@@ -225,21 +225,27 @@ int SeriesAddSample(Series *series, api_timestamp_t timestamp, double value) {
 }
 
 SeriesIterator SeriesQuery(Series *series, api_timestamp_t minTimestamp, api_timestamp_t maxTimestamp) {
-    SeriesIterator iter;
+    SeriesIterator iter = { 0 };
     timestamp_t rax_key;
     iter.series = series;
+    ChunkFuncs *funcs = iter.series->funcs;
+    
     // get the rightmost chunk whose base timestamp is smaller or equal to minTimestamp
     seriesEncodeTimestamp(&rax_key, minTimestamp);
     iter.dictIter = RedisModule_DictIteratorStartC(series->chunks, "<=", &rax_key, sizeof(rax_key));
+    RedisModule_DictNextC(iter.dictIter, NULL, (void*)&iter.currentChunk);
```

Comment:
@ashtul don't you want to check the result and/or if currentChunk is the first chunk we need to iterate over?
What's the problem with the previous solution?

Reply:
The previous code with reseek was redundant. 
`RedisModule_DictIteratorReseekC` and `RedisModule_DictIteratorStartC` are identical inthis context.
The previous solution check if the chunk is the right one every time we cal `GetNext`. In this solution, we check once during initilization and the, on calling `GetNext`, you can call `ChunkIteratorGetNext` right away instead of checking the chunk first. Only if call returns CR_END we check for next chunk etc'.